autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/UpdateFailedHandler.java

```diff
@@ -51,15 +59,35 @@ public void onApplicationEvent(UpdateFailedEvent event) {
         if (failed < RETRY_THRESHOLD) {
             updateFailures.put(autoscaleClusterId, failed);
             LOGGER.debug("Increased failed count '{}' for cluster '{}'", failed, cluster.getStackCrn());
+            // TODO: Determine what needs to be done if ForbiddenException occurs with endUser.
+            if (event.getCausedBy() instanceof ForbiddenException) {
+                Integer forbiddencount = getForbiddenFailureCount(autoscaleClusterId) + 1;
```

Comment:
Get rid of forbiddenCountFailure altogether. Tracking it here means we end up with different state here as compared to the DB, and if there is a successful attempt - this class still tracks previous failures.

So - get this information from the current DB state, passed in via the event.